### PR TITLE
chore(bigquery): wrap go errors

### LINKED ISSUE #2700: bigquery: audit existing samples/quickstarts for error wrapping best practices
Our minimum supported version for go is now ensures all clients should respect error wrapping format directives.  Let's audit all our existing BigQuery samples and ensure they do things correctly with respect introspecting error details.

ex: instances of `fmt.Errorf("blah: %v", err)` should likely be rewritten as `fmt.Errorf("blah: %w", err)`

### PR DESCRIPTION
**Pull request by alvarowolfx**
Resolves #2700

### CODE REVIEW

File: bigquery/simpleapp/simpleapp.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func main() {
 
 	client, err := bigquery.NewClient(ctx, projectID)
 	if err != nil {
-		log.Fatalf("bigquery.NewClient: %v", err)
+		log.Fatalf("bigquery.NewClient: %w", err)
```
Comment (quartzmo) [inline bigquery/simpleapp/simpleapp.go:45]: The `%w` is not supposed to do anything special in `Fatalf` (`Printf`), so no need to change these usages.

Comment (quartzmo) [reply bigquery/simpleapp/simpleapp.go:45]: "log.Fatalf does not support error-wrapping directive %w" - linter in VS Code

Comment (alvarowolfx) [reply bigquery/simpleapp/simpleapp.go]: oops, good catch. That was a mistake that was undone locally and when I came back to the computer I saw that the file was not saved 🤦

File: bigquery/snippets/models/bigquery_get_model.go [hunk 2]
```diff
@@ -32,13 +32,13 @@ func printModelInfo(w io.Writer, projectID, datasetID, modelID string) error {
...
 	if err != nil {
-		return fmt.Errorf("bigquery.NewClient: %v", err)
+		return fmt.Errorf("bigquery.NewClient: %w", err)
 	}
 	defer client.Close()
 
 	meta, err := client.Dataset(datasetID).Model(modelID).Metadata(ctx)
 	if err != nil {
-		return fmt.Errorf("Metadata: %v", err)
+		return fmt.Errorf("couldn't retrive metadata: %v", err)
```
Comment (alvarowolfx) [inline bigquery/snippets/models/bigquery_get_model.go:41]: typo on error message

File: bigquery/snippets/table/bigquery_update_materialized_view.go [hunk 3]
```diff
@@ -31,15 +31,15 @@ func updateMaterializedView(projectID, datasetID, viewID string) error {
...
+		return fmt.Errorf("bigquery.NewClient: %w", err)
 	}
 	defer client.Close()
 
 	// Retrieve current view metadata.
 	viewRef := client.Dataset(datasetID).Table(viewID)
 	meta, err := viewRef.Metadata(ctx)
 	if err != nil {
-		return fmt.Errorf("Metadata(): %v", err)
+		return fmt.Errorf("Metadata(): %w", err)
```
Comment (alvarowolfx) [inline bigquery/snippets/table/bigquery_update_materialized_view.go:42]: the error message can be improved

**Review by shollyman — APPROVED** [review]
This PR brought to you by the letters "v" and "w".  Thanks for auditing the samples!

**Review by muncus — APPROVED** [review]

**Pull request merged**